### PR TITLE
Add SSO via OAuth2.0

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Add a module for authorization and authentification using OIDC and OAuth2.0 protocol.

Please delete either the [master < EPIC] or [master < Task] part, depending on what are your needs.

[master < Epic] PR
- [x] Write E2E tests
- [x] Compare the [benchmarking results](https://bench-graph.memgraph.com/) between the master branch and the Epic branch
- [ ] Provide the full content or a guide for the final git message
    - [FINAL GIT MESSAGE]

[master < Task] PR
- [ ] Provide the full content or a guide for the final git message
    - **[FINAL GIT MESSAGE]**

### CI Testing Labels
Please select the appropriate CI test labels _(CI -build=**build-name** -test=**test-suite**)_

### Documentation checklist
- [x] Add the documentation label tag
- [x] Add the bug / feature label tag
- [x] Add the milestone for which this feature is intended
    - If not known, set for a later milestone
- [x] Write a release note, including added/changed clauses
    - **[Add module for enterprise authentification and authorization with OIDC and OAuth2.0 protocols.]**
- [x] Link the documentation PR here
    - **[[Documentation PR link](https://github.com/memgraph/documentation/pull/876)]**
- [x] Tag someone from docs team in the comments

### CODE REVIEW

File: src/auth/reference_modules/oauth.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+        config["user_info_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_USER_INFO_URL", "")
+        config["client_id"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CLIENT_ID", "")
+        config["client_secret"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CLIENT_SECRET", "")
+        config["callback_url"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_CALLBACK_URL", "")
+        config["scope"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_SCOPE", "")
+        config["tenant_id"] = os.environ.get("AUTH_OAUTH_ENTRA_ID_TENANT_ID", "")
+        config["role_mapping"] = _load_role_mappings(os.environ.get("AUTH_OAUTH_ENTRA_ID_ROLE_MAPPING", ""))
+
+    elif scheme == "oauth-okta":
+        config["id_issuer"] = os.environ.get("AUTH_OAUTH_OKTA_ISSUER", "")
```
Comment (gitbuda) [inline src/auth/reference_modules/oauth.py:80]: let's change all env vars to be prefixed with MEMGRAPH_

Comment (antepusic) [reply src/auth/reference_modules/oauth.py:80]: Done in #2049. @imilinovic You may merge the changes from that PR’s branch :point_up:

**Comment by kgolubic** [comment]
@imilinovic I'm missing the Release note entry and documentation PR.

**Review by antepusic — APPROVED** [review]
I think this one is good to go

**Comment by antepusic** [comment]
@kgolubic I think this PR and #2049 should have a joint release note because they go hand-in-hand: #2049 adds SSO for Entra ID and Okta over the SAML protocol, whereas this PR does the same with OIDC/Oauth

**Pull request merged**